PetrolEngine.cpp

```diff
@@ -1,20 +1,34 @@
 #include "PetrolEngine.hpp"
+#include "CarExceptions.hpp"
 #include <iostream>
+#include <stdlib.h>
 
-PetrolEngine::PetrolEngine(int power, float capacity, int gears)
-    : power_(power)
-    , capacity_(capacity)
-    , gears_(gears)
-    , currentGear_(0)
-{
-    std::cout << __FUNCTION__ << std::endl;
+PetrolEngine::PetrolEngine(Power power, Capacity capacity, Gears gears)
+    : power_(power), capacity_(capacity), gears_(gears), currentGear_(0) {
+  std::cout << __FUNCTION__ << ": " << __FILE__ << ": " << __LINE__
+            << std::endl;
 }
 
-void PetrolEngine::changeGear(int gear)
-{
-    // TODO: Add checking if gear is between -1 and gears_
-    // -1 is for REAR
-    // 0 is for NEUTRAL
-    currentGear_ = gear;
-    std::cout << __FUNCTION__ << std::endl;
+void PetrolEngine::changeGear(int gear) {
+  // TODO: Add checking if gear is between -1 and gears_
+  // -1 is for REAR
+  // 0 is for NEUTRAL
+
+  std::cout << __FUNCTION__ << ": " << __FILE__ << ": " << __LINE__
+            << std::endl;
+  if (abs(gear - currentGear_) != 1 or gear < -1 or gear > gears_.gears_) {
+    throw InvalidGear("Zgrzyyyyt!");
+  }
```

Comment:
I suggest making '-1' and '1' constexprs instead of magic values.

Reply:
+1